phrosty/pipeline.py

```diff
@@ -65,14 +67,28 @@ def __init__( self, imagepath, pointing, sca ):
 
         self.pointing = pointing
 
+        # Intermediate files
+        if self.keep_intermediate:
+            self.skysub_path = None
+            self.detmask_path = None
+            self.input_sci_psf_path = None
+            self.input_templ_psf_path = None
+            self.aligned_sci_img_path = None
+            self.aligned_templ_img_path = None
+            self.aligned_sci_psf_path = None
+            self.aligned_templ_psf_path = None
+            self.crossconv_sci_path = None
+            self.crossconv_templ_path = None
+            self.decorr_kernel_path = None
+
+        # Always save and output these
         self.decorr_psf_path = {}
         self.decorr_zptimg_path = {}
         self.decorr_diff_path = {}
         self.zpt_stamp_path = {}
         self.diff_stamp_path = {}
 
-        self.skysub_path = None
-        self.detmas_path = None
+        # Used internally, held in-memory
```

Comment:
I would say "held in memory".  No need for a hyphen.  I don't think you're modifying anything after "in-memory".